test/colab/colab-test.mk

```diff
@@ -0,0 +1,31 @@
+.PHONY: colab-test-internal
+colab-test-internal: $(TEST_TOKEN_FILE) $(COLAB_TEST_HOME)/test_import_login_init.ipynb colab-test-pull
+	$(DOCKER_RUN)
+
+.PHONY: colab-test-internal-local
+colab-test-internal-local: $(TEST_TOKEN_FILE) $(COLAB_TEST_HOME)/test_import_login_init.ipynb colab-test-build
+	$(DOCKER_RUN)
+
+.PHONY: colab-test-pull
+colab-test-pull:
+	@docker pull $(DOCKER_IMAGE_NAME)
+
+.PHONY: colab-test-build
+colab-test-build: $(COLAB_IMAGE_BUILD_STAMP)
+$(COLAB_IMAGE_BUILD_STAMP): $(COLAB_TEST_HOME)/requirements-fixed.txt test/colab/Dockerfile
```

Comment:
Only build once as it's time consuming, unless the Dockerfile or requirements have changed, in which case we rebuild it.